packages/react/src/components/Modal/Modal.base.tsx

```diff
@@ -413,11 +413,11 @@ export const ModalBase: React.FunctionComponent<IModalProps> = React.forwardRef<
         elementToFocusOnDismiss={elementToFocusOnDismiss}
         isClickableOutsideFocusTrap={isModeless || isClickableOutsideFocusTrap || !isBlocking}
         ignoreExternalFocusing={ignoreExternalFocusing}
-        forceFocusInsideTrap={isModeless ? !isModeless : forceFocusInsideTrap}
+        forceFocusInsideTrap={forceFocusInsideTrap && !isModeless}
         firstFocusableSelector={firstFocusableSelector}
         focusPreviouslyFocusedInnerElement
         onBlur={internalState.isInKeyboardMoveMode ? handleExitKeyboardMoveMode : undefined}
-        enableAriaHiddenSiblings={enableAriaHiddenSiblings}
+        // enableAriaHiddenSiblings is handled by the Popup
```

Comment:
This part is important: only have one component trying to handle sibling hiding. In this case use Popup since it's the outer component and it has a more refined mechanism for determining whether to hide siblings (checking 
 `aria-modal`).